tests/gis_tests/gdal_tests/test_raster.py

```diff
@@ -470,6 +471,61 @@ def test_raster_warp_nodata_zone(self):
         # The result is an empty raster filled with the correct nodata value.
         self.assertEqual(result, [23] * 16)
 
+    def test_raster_clone(self):
+        source = GDALRaster({
+            'datatype': 1,
+            'driver': 'MEM',
+            'width': 4,
+            'height': 4,
+            'srid': 3086,
+            'origin': (500000, 400000),
+            'scale': (100, -100),
+            'skew': (0, 0),
+            'bands': [{
+                'data': range(16),
+                'nodata_value': 23,
+            }],
+        })
+
+        clone = source.clone()
+
+        # Test cloned raster has same attributes
+        self.assertEqual(source.name, clone.name)
+        self.assertEqual(source.srs.srid, clone.srs.srid)
+        self.assertEqual(source.width, clone.width)
+        self.assertEqual(source.height, clone.height)
+        self.assertAlmostEqual(source.origin[0], clone.origin[0], 3)
+        self.assertAlmostEqual(source.origin[1], clone.origin[1], 3)
+        self.assertAlmostEqual(source.scale[0], clone.scale[0], 3)
+        self.assertAlmostEqual(source.scale[1], clone.scale[1], 3)
+        self.assertEqual(source.skew, clone.skew)
```

Comment:
Can we use instead `assertEqual()` and `assertIsNot()`?
```suggestion
        self.assertEqual(clone, source)
        self.assertIsNot(clone, source)
```

Reply:
__eq__ is not implemented
If you use copy_ds() you can't use assertEqual